spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -1461,12 +1463,41 @@ private Object createHttpServer(File outputDir) {
             .getConstructor(new Class[]{
               File.class, SecurityManager.class, int.class, String.class});
         return constructor.newInstance(new Object[] {
-          outputDir, new SecurityManager(conf), 0, "HTTP Server"});
+          outputDir, createSecurityManager(conf), 0, "HTTP Server"});
       } catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException |
           InstantiationException | InvocationTargetException e1) {
         logger.error(e1.getMessage(), e1);
         return null;
       }
     }
   }
+
+  /**
+   * Constructor signature of SecurityManager changes in spark 2.10, so we use this method to create
+   * Security properly for different versions of spark
+   *
+   * @param conf
+   * @return
+   * @throws ClassNotFoundException
+   * @throws NoSuchMethodException
+   * @throws IllegalAccessException
+   * @throws InvocationTargetException
+   * @throws InstantiationException
+   */
+  private Object createSecurityManager(SparkConf conf) throws ClassNotFoundException,
+      NoSuchMethodException, IllegalAccessException, InvocationTargetException,
+      InstantiationException {
+    Object securityManager = null;
+    try {
+      Constructor<?> smConstructor = getClass().getClassLoader()
+          .loadClass("org.apache.spark.SecurityManager")
+          .getConstructor(new Class[]{ SparkConf.class });
```

Comment:
Since the two-arg constructor is from the newest version of Spark, and the one-arg constructor is from the older version of Spark, should we look for the two-arg constructor first? That way you'd only have to look for both constructors if you're using an older version of Spark.

Then again this is probably negligible because it looks like this method would only ever be called once per Spark interpreter anyway.